SalesforceFoundation/EDA
# [EDA] Lead Conversion Account Naming Setting

# Changes
## Updates for Lead Conversion
We added a new EDA Setting called Automatically Rename Lead-Converted Accounts. Now, admins can choose to automatically rename Accounts created when a Lead is converted to use the account naming format that's associated with its record type. Without this setting, the Account name is determined by the Company field on the Lead record. 

For example, let’s say you selected the Administrative record type in Automatically Rename Lead-Converted Accounts. When you convert a Lead and specify the Administrative record type, the Account record is renamed to use the Administrative Account Name Format from EDA Settings. An Account for a Lead named Louis Lombard is automatically renamed Lombard Administrative Account.

Starting with the trial version of EDA 1.xxx, this setting is configured to specify the Administrative and Household Account record types by default. If you started with a version of EDA earlier than 1.xxx or you're starting with EDA 1.xxx via the installer rather than the trial, no default record types are specified.

For more information about working with Leads, see [Lead](https://powerofus.force.com/s/article/EDA-Lead) in the EDA Documentation. 

# Issues Closed
[[W-9020537]](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008xk4tIAA/view)

# New Metadata
## Custom Labels
- stgLeadConversionAccountNamingTitle
- stgLeadConversionAccountNaming
- stgLeadConversionAccountNamingDesc

## Apex
- AccountModelSettingsController
- LeadConversionAccountNamingMapper
- LeadConversionAccountNamingMapper_TEST
- LeadConversionAccountNamingModel
- LeadConversionAccountNamingModel_TEST
- LeadConversionAccountNamingService
- LeadConversionAccountNamingService_TEST
- LeadConversionAccountNamingVMapper
- LeadConversionAccountNamingVMapper_TEST

# Testing Notes

* Verify when the EDA Settings come up (New EDA Settings QA Only), the Lead Conversion Auto Account Naming comes up after Rename Household Accounts
* Add new Account Record Types and verify they show up in the Available Record Types
* Select 1 or more Record Types and verify they move over to the Selected Account Types
* Click Save, refresh the page, and verify the Selected Record Types are still there
* Remove 1 or more Record Types from the Selected Account Record Types
* Click Save, refresh the page, and verify the Records are Removed and moved to the Available Account Record Types
* Go to Setup/Custom Settings, and Select Hierarchy Settings
* Verify the Lead Naming Account Record Types field is populated with the Selected Record Types
* Verify all of the existing functionality within Account Model are still working as expected, pay particular attention to the Accounts Without Contacts Section.
